pkg/fqdn/service/service_test.go

```diff
@@ -0,0 +1,244 @@
+	health "github.com/cilium/cilium/pkg/hive/health/types"
+	"github.com/cilium/cilium/pkg/ipcache"
+	"github.com/cilium/cilium/pkg/logging"
+	"github.com/cilium/cilium/pkg/option"
+	"github.com/cilium/cilium/pkg/policy"
+	"github.com/cilium/cilium/pkg/policy/api"
+	"github.com/cilium/cilium/pkg/testutils"
+	testidentity "github.com/cilium/cilium/pkg/testutils/identity"
+	"github.com/cilium/cilium/pkg/time"
+
+	pb "github.com/cilium/cilium/api/v1/standalone-dns-proxy"
+)
+
+func TestFQDNDataServer(t *testing.T) {
+	// testutils.PrivilegedTest(t)
+
+	test := map[string]struct {
+		port                     int
+		enableL7Proxy            bool
+		enableStandaloneDNSProxy bool
+		serverPort               int
+		err                      error
+	}{
+		"Successfully running the server": {
+			port:                     1234,
+			enableL7Proxy:            true,
+			enableStandaloneDNSProxy: true,
+			serverPort:               40045,
+			err:                      nil,
+		},
+		"Failure on running the server due to invalid server port": {
+			port:                     1234,
+			serverPort:               0,
+			enableStandaloneDNSProxy: true,
+			enableL7Proxy:            true,
+			err:                      errors.New("listen tcp: address -1: invalid port"),
+		},
+	}
+
+	for scenario, tt := range test {
+		t.Run(scenario, func(t *testing.T) {
+
+			em := endpointmanager.New(&dummyEpSyncher{}, nil, nil, nil, nil)
+			pr := policy.NewPolicyRepository(hivetest.Logger(t), nil, nil, nil, nil, api.NewPolicyMetricsNoop())
+			ipc := ipcache.NewIPCache(&ipcache.Configuration{
+				Context:           context.TODO(),
+				IdentityAllocator: testidentity.NewMockIdentityAllocator(nil),
+				PolicyHandler:     pr.GetSelectorCache(),
+				DatapathHandler:   em,
+			})
+			nm := namemanager.New(namemanager.ManagerParams{
+				Config: namemanager.NameManagerConfig{
+					MinTTL:            1,
+					DNSProxyLockCount: defaults.DNSProxyLockCount,
+					StateDir:          defaults.StateDir,
+				},
+				IPCache: ipc,
+			})
+
+			hive := hive.New(
+				Cell,
+				cell.Provide(func() *ipcache.IPCache {
+					return ipc
+				}),
+				cell.Provide(func() endpointmanager.EndpointManager {
+					return em
+				}),
+				cell.Provide(func() namemanager.NameManager { return nm }),
+				cell.Provide(func(lc cell.Lifecycle, p health.Provider, jr job.Registry) job.Group {
+					h := p.ForModule(cell.FullModuleID{"test"})
+					jg := jr.NewGroup(h)
+					lc.Append(jg)
+					return jg
+				}),
+				cell.Provide(func(lc cell.Lifecycle) messagehandler.DNSRequestHandler {
+					return messagehandler.NewDNSRequestHandler(
+						messagehandler.DNSRequestHandlerParams{
+							Lifecycle:         lc,
+							Logger:            hivetest.Logger(t),
+							NameManager:       nm,
+							ProxyInstance:     nil,
+							ProxyAccessLogger: nil,
+						})
+				}),
+				cell.Invoke(newServer),
+			)
+
+			flags := pflag.NewFlagSet("", pflag.ContinueOnError)
+			hive.RegisterFlags(flags)
```

Comment:
`hive.AddConfigOverride` is there for this purpose and it's a bit safer to use (no chance of typoing the flags):
```
hive.AddConfigOverride(
  testHive,
  func(cfg *Config) {
    cfg.EnableStandaloneDNSProxy = tt.enableStandaloneDNSProxy
  })
```